packages/upload-api/src/executor.js

```diff
@@ -0,0 +1,44 @@
+// @ts-nocheck
+import * as API from '../test/types.js'
+import * as Server from '@ucanto/server'
+
+/**
+ * @typedef {object} Context
+ * @property {Store<Uint8Array, API.AgentMessage>} messageStore
+ */
+
+/**
+ * @template K, V
+ * @typedef {object} Store
+ * @property {(data: V) => Promise<API.Result<{}, Error>>} put
+ */
+
+/**
+ * @template {Record<string, any>} T
+ * @param {API.AgentMessage} input
+ * @param {API.Server<T>} server
+ * @param {Context} context
+ * @returns {Promise<API.Result<API.AgentMessage<T>, Error>>}
+ */
+export const execute = async (input, server, context) => {
+  const saveInput = await context.messageStore.put(input)
```

Comment:
What is the difference between this and `agentStore.messages`?

Reply:
`agentStore.messages` is not part of the `AgentMessage` interface, it's merely an implementation detail of the in-memory `AgentStore` (which in fact returns `this`). Different implementation e.g. w3infra may not have `.messages` field.

Reply:
Oh wait never mind this file was not meant to be committed it was a sketch that got integrated into `./lib.js` instead.